### PR TITLE
Cleanup xrdutil

### PR DESCRIPTION
**Pull request by ZackAttack614**
Clean up the `xrdutil.py` file.

- Add formatting with [black](https://pypi.org/project/black/).
- Add type annotations.
- Remove `USE_NUMBA` checks.
- Remove unnecessary functions and variables.
- Add `@deprecated` decorator to warn users of functions marked for sunset.

### CODE REVIEW

File: hexrd/xrdutil/utils.py [hunk 1]
```diff
@@ -91,11 +95,11 @@ def __init__(self, ome_eta_archive):
         self.etas = ome_eta['etas']
         self.omegas = ome_eta['omegas']
 
-    def save(self, filename):
+    def save(self, filename: str) -> None:
         self.save_eta_ome_maps(self, filename)
 
     @staticmethod
-    def save_eta_ome_maps(eta_ome, filename):
+    def save_eta_ome_maps(eta_ome, filename: str) -> None:
```
Comment (kevindlewis23) [inline hexrd/xrdutil/utils.py:102]: Missing type annotation on eta_ome

File: hexrd/deprecation.py [hunk 2]
```diff
@@ -21,6 +21,7 @@ def wrapper(*args, **kwargs):
                 print(
                     f"Warning: {func.__name__} is deprecated and is marked for"
                     f" removal. Please use {new_func} instead."
+                    f" Deprecation date: {deprecation_date}"
```
Comment (psavery) [inline hexrd/deprecation.py:24]: Can you say `Removal date:` instead? I think "Deprecation date" sounds like the day it was deprecated.

File: hexrd/xrdutil/utils.py [hunk 3]
```diff
@@ -1101,72 +1101,78 @@ def simulateGVecs(pd, detector_params, grain_params,
...
+            minEnergy
+        ), 'energy cutoff ranges must have the same length'
         multipleEnergyRanges = True
-        lmin = []
-        lmax = []
-        for i in range(len(maxEnergy)):
-            lmin.append(processWavelength(maxEnergy[i]))
-            lmax.append(processWavelength(minEnergy[i]))
+        lmin = [processWavelength(e) for e in minEnergy]
+        lmax = [processWavelength(e) for e in maxEnergy]
```
Comment (psavery) [inline hexrd/xrdutil/utils.py:1167]: It looks like `minEnergy` and `maxEnergy` got swapped in the change here. Can you change them back?

Comment (ZackAttack614) [reply hexrd/xrdutil/utils.py]: Resolved

**Review by psavery — APPROVED** [review]
LGTM.

**Pull request merged**